widgets/views/wallEntry.php

```diff
@@ -62,21 +62,25 @@
     <?php if ($calendarEntry->canRespond()): ?>
        <div class="row" style="padding-top:10px">
             <div class="col-md-12">
-                <?= Button::defaultType(Yii::t('CalendarModule.views_entry_view', "Attend"))->sm()
-                    ->icon($participantSate === CalendarEntryParticipant::PARTICIPATION_STATE_ACCEPTED ? 'fa-check' : null)
-                    ->action('calendar.respond', $contentContainer->createUrl('/calendar/entry/respond', ['type' => CalendarEntryParticipant::PARTICIPATION_STATE_ACCEPTED, 'id' => $calendarEntry->id]))?>
 
+                <?php
+                $pButton = function($state, $label) use ($calendarEntry, $participantSate, $contentContainer) {
```

Comment:
`$pButton` is not very descriptive without reading the code. Ideally this should be implemented as subclass of Button, since code within a view files should be avoided if possible.